xbmc/cores/VideoPlayer/DVDCodecs/Video/DXVA.cpp

```diff
@@ -793,21 +798,141 @@ HRESULT CVideoBufferShared::GetResource(ID3D11Resource** ppResource)
   {
     // open resource on app device
     ComPtr<ID3D11Device> pD3DDevice = DX::DeviceResources::Get()->GetD3DDevice();
-    hr = pD3DDevice->OpenSharedResource(handle, __uuidof(ID3D11Resource), &m_sharedRes);
+    if (FAILED(hr = pD3DDevice->OpenSharedResource(handle, __uuidof(ID3D11Resource), &m_sharedRes)))
+    {
+      CLog::LogF(LOGDEBUG, "unable to open the shared resource, error description: {}",
+                 DX::GetErrorDescription(hr));
+      return hr;
+    }
+  }
+
+  if (FAILED(hr = m_sharedRes.CopyTo(ppResource)))
```

Comment:
Now seems more logic move this code at end of method (when all is initialized) and after "wait"

```c++
hr = m_sharedRes.CopyTo(ppResource);

return hr;
```

Reply:
it's a pointer copy, but OK.

moving the fence initialization in the if (m_sharedRes) block since there is no need to log errors for each picture if something is wrong in the init.